### ISSUE TITLE
Slider violates start and end when using keyboard navigation

### ISSUE THREAD

**Issue by gbrunick** [bokeh/bokeh#7622]
You can see this behavior on the official example:

[http://bokeh.pydata.org/en/latest/docs/gallery/slider.html](http://bokeh.pydata.org/en/latest/docs/gallery/slider.html)

Drag the amplitude slider with the mouse, and it varies from 0.1 to 10.  Use the keyboard instead and the slider appears to vary from 0 to 10.1 exceeding the start bound of 0.1 and the end bound 10.  This happens in Bokeh 0.12.14.

The fix appears to be:

```
--- bokeh-widgets-0.12.14.js
+++ bokeh-widgets-0.12.14.js
@@ -225,11 +225,11 @@ function _(require, module, exports) {
                     var value = Number(_this.sliderEl.noUiSlider.get());
                     switch (e.which) {
                     case 37: {
-                            value -= step;
+                            value = Math.max(value - step, start);
                             break;
                         }
                     case 39: {
-                            value += step;
+                            value = Math.min(value + step, end);
                             break;
                         }
                     default:
```

**Comment by bryevdv** [comment]
@gbrunick I thought this might be a dupe of #7349 since in principle `noUISlider` should prevent this. But if a little defensive coding also fixes, I am happy to include. Can you make the small PR? The location of that code in the source is here:

https://github.com/bokeh/bokeh/blob/master/bokehjs/src/coffee/models/widgets/abstract_slider.ts#L95

**Comment by gbrunick** [comment]
Sure.  I will put it in a PR, but it will probably not happen before the weekend.

**Comment by bryevdv** [comment]
@gbrunick FYI we are planning to release early this week, so if you do have an opportunity to work on it this before then, we'd be happy to get it in the next release.

**Comment by gbrunick** [comment]
Done.  PR #7649
